diff --git a/source/gui/FreeTypeGX.cpp b/source/gui/FreeTypeGX.cpp
index bab0836..38dd74d 100644
--- a/source/gui/FreeTypeGX.cpp
+++ b/source/gui/FreeTypeGX.cpp
@@ -189,7 +189,7 @@ ftgxCharData * FreeTypeGX::cacheGlyphData(wchar_t charCode, int16_t pixelSize) {
     }
 
     gIndex = FT_Get_Char_Index(ftFace, (FT_ULong) charCode);
-    if (gIndex != 0 && FT_Load_Glyph(ftFace, gIndex, FT_LOAD_DEFAULT | FT_LOAD_RENDER) == 0) {
+    if (gIndex != 0 && FT_Load_Glyph(ftFace, gIndex, FT_LOAD_TARGET_LIGHT | FT_LOAD_RENDER) == 0) {
         if (ftFace->glyph->format == FT_GLYPH_FORMAT_BITMAP) {
             FT_Bitmap *glyphBitmap = &ftFace->glyph->bitmap;
 
