--- a/source/video/shaders/Shader3D.cpp	2021-04-23 11:15:05.587960502 +0200
+++ b/source/video/shaders/Shader3D.cpp	2021-04-23 11:20:17.018714286 +0200
@@ -216,13 +216,13 @@
     projectionMatrixLocation = 16;
     viewMatrixLocation = 32;
     vertexShader.addUniformVar((GX2UniformVar) {
-        "modelMatrix", GX2_SHADER_VAR_TYPE_MATRIX4X4, 1, modelMatrixLocation, -1
+        "modelMatrix", GX2_SHADER_VAR_TYPE_FLOAT4X4, 1, modelMatrixLocation, -1
     });
     vertexShader.addUniformVar((GX2UniformVar) {
-        "viewMatrix", GX2_SHADER_VAR_TYPE_MATRIX4X4, 1, projectionMatrixLocation, -1
+        "viewMatrix", GX2_SHADER_VAR_TYPE_FLOAT4X4, 1, projectionMatrixLocation, -1
     });
     vertexShader.addUniformVar((GX2UniformVar) {
-        "projectionMatrix", GX2_SHADER_VAR_TYPE_MATRIX4X4, 1, viewMatrixLocation, -1
+        "projectionMatrix", GX2_SHADER_VAR_TYPE_FLOAT4X4, 1, viewMatrixLocation, -1
     });
 
     positionLocation = 0;
--- a/source/video/shaders/ShaderFractalColor.cpp	2021-04-23 11:15:05.587960502 +0200
+++ b/source/video/shaders/ShaderFractalColor.cpp	2021-04-23 11:19:35.077266135 +0200
@@ -308,13 +308,13 @@
     projectionMatrixLocation = 16;
     viewMatrixLocation = 32;
     vertexShader.addUniformVar((GX2UniformVar) {
-        "modelMatrix", GX2_SHADER_VAR_TYPE_MATRIX4X4, 1, modelMatrixLocation, -1
+        "modelMatrix", GX2_SHADER_VAR_TYPE_FLOAT4X4, 1, modelMatrixLocation, -1
     });
     vertexShader.addUniformVar((GX2UniformVar) {
-        "projectionMatrix", GX2_SHADER_VAR_TYPE_MATRIX4X4, 1, projectionMatrixLocation, -1
+        "projectionMatrix", GX2_SHADER_VAR_TYPE_FLOAT4X4, 1, projectionMatrixLocation, -1
     });
     vertexShader.addUniformVar((GX2UniformVar) {
-        "viewMatrix", GX2_SHADER_VAR_TYPE_MATRIX4X4, 1, viewMatrixLocation, -1
+        "viewMatrix", GX2_SHADER_VAR_TYPE_FLOAT4X4, 1, viewMatrixLocation, -1
     });
 
     positionLocation = 0;
