diff --git a/source/video/CVideo.cpp b/source/video/CVideo.cpp
index 4eb8ad2..b1fd1d5 100644
--- a/source/video/CVideo.cpp
+++ b/source/video/CVideo.cpp
@@ -104,16 +104,12 @@ CVideo::CVideo(int32_t forceTvScanMode, int32_t forceDrcScanMode) {
     //! this should be ok for our purpose i guess
 
     //! Setup TV depth buffer (can be the same for both if rendered one after another)
-    uint32_t size, align;
     GX2InitDepthBuffer(&tvDepthBuffer, GX2_SURFACE_DIM_TEXTURE_2D, tvColorBuffer.surface.width, tvColorBuffer.surface.height, 1, GX2_SURFACE_FORMAT_FLOAT_R32, (GX2AAMode)tvAAMode);
     tvDepthBuffer.surface.image = MEM1_alloc(tvDepthBuffer.surface.imageSize, tvDepthBuffer.surface.alignment);
     GX2Invalidate(GX2_INVALIDATE_MODE_CPU, tvDepthBuffer.surface.image, tvDepthBuffer.surface.imageSize);
 
     //! Setup TV HiZ buffer
-    GX2CalcDepthBufferHiZInfo(&tvDepthBuffer, &size, &align);
-    tvDepthBuffer.hiZPtr = MEM1_alloc(size, align);
-    GX2Invalidate(GX2_INVALIDATE_MODE_CPU, tvDepthBuffer.hiZPtr, size);
-    GX2InitDepthBufferHiZEnable(&tvDepthBuffer, GX2_ENABLE);
+    GX2InitDepthBufferHiZEnable(&tvDepthBuffer, GX2_DISABLE);
 
     //! Setup color buffer for DRC rendering
     GX2InitColorBuffer(&drcColorBuffer, GX2_SURFACE_DIM_TEXTURE_2D, 854, 480, 1, GX2_SURFACE_FORMAT_UNORM_R8_G8_B8_A8, (GX2AAMode)tvAAMode);
@@ -203,7 +199,6 @@ CVideo::~CVideo() {
     MEM1_free(drcColorBuffer.surface.image);
     //! free depth buffers
     MEM1_free(tvDepthBuffer.surface.image);
-    MEM1_free(tvDepthBuffer.hiZPtr);
     //! free context buffers
     MEM2_free(tvContextState);
     MEM2_free(drcContextState);
