diff --git a/source/gui/system/SDLSystem.cpp b/source/gui/system/SDLSystem.cpp
index 23dfe95..cf2ff99 100644
--- a/source/gui/system/SDLSystem.cpp
+++ b/source/gui/system/SDLSystem.cpp
@@ -21,7 +21,10 @@
 #include <gui/system/libgui_log.h>
 
 SDLSystem::SDLSystem() {
-    SDL_Init(SDL_INIT_AUDIO | SDL_INIT_VIDEO);
+    if (SDL_Init(SDL_INIT_AUDIO | SDL_INIT_VIDEO) != 0) {
+        LG_Log("SDL init error: %s\n", SDL_GetError());
+        return;
+    }
 
     auto SDLFlags = SDL_RENDERER_ACCELERATED | SDL_RENDERER_PRESENTVSYNC;
 
@@ -43,11 +46,6 @@ SDLSystem::SDLSystem() {
         return;
     }
 
-    if (SDL_Init(SDL_INIT_AUDIO) != 0) {
-        LG_Log("SDL init error: %s\n", SDL_GetError());
-        return;
-    }
-
     int flags = 0;
     int result = 0;
     if (flags != (result = Mix_Init(flags))) {
@@ -55,7 +53,7 @@ SDLSystem::SDLSystem() {
         LG_Log("Mix_Init: %s\n", Mix_GetError());
     }
 
-    auto dev = Mix_OpenAudio(22050, AUDIO_S16SYS, 2, 640);
+    auto dev = Mix_OpenAudio(22050, AUDIO_S16MSB, 2, 640);
     SDL_PauseAudioDevice(dev, 0);
 
     TTF_Init();
