diff a/source/gui/system/SDLSystem.cpp b/source/gui/system/SDLSystem.cpp
--- a/source/gui/system/SDLSystem.cpp	2021-12-17 15:25:42.930315825 +0100
+++ b/source/gui/system/SDLSystem.cpp	2021-12-17 15:25:56.788791663 +0100
@@ -26,7 +26,7 @@
         return;
     }
 
-    auto SDLFlags = SDL_RENDERER_ACCELERATED | SDL_RENDERER_PRESENTVSYNC;
+    auto SDLFlags = SDL_RENDERER_ACCELERATED;
 
     //Setup window
     window = SDL_CreateWindow(nullptr, SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED, 1280, 720, 0);
