diff --git a/include/gui-sdl/gui/GuiImage.h b/include/gui-sdl/gui/GuiImage.h
index 923e8fd..ff0fbea 100644
--- a/include/gui-sdl/gui/GuiImage.h
+++ b/include/gui-sdl/gui/GuiImage.h
@@ -51,5 +51,5 @@ private:
     SDL_Color color = {0, 0, 0, 0};
 
 protected:
-    SDL_BlendMode blendMode;
+    SDL_BlendMode blendMode = SDL_BLENDMODE_NONE;
 };
diff --git a/include/gui-sdl/gui/GuiTextureData.h b/include/gui-sdl/gui/GuiTextureData.h
index 0690b0c..b4bea96 100644
--- a/include/gui-sdl/gui/GuiTextureData.h
+++ b/include/gui-sdl/gui/GuiTextureData.h
@@ -37,5 +37,5 @@ protected:
     SDL_Texture *texture = nullptr;
     int32_t width = 0;
     int32_t height = 0;
-    SDL_BlendMode blendMode;
+    SDL_BlendMode blendMode = SDL_BLENDMODE_NONE;
 };
