diff --git a/include/gui-sdl/gui/GuiImage.h b/include/gui-sdl/gui/GuiImage.h
index 8a4e54d..923e8fd 100644
--- a/include/gui-sdl/gui/GuiImage.h
+++ b/include/gui-sdl/gui/GuiImage.h
@@ -41,10 +41,15 @@ public:
 
     void setTexture(GuiTextureData *tex);
 
+    int setBlendMode(SDL_BlendMode blendMode);
+
 private:
     GuiTextureData *texture = nullptr;
     bool freeTextureData = false;
 
     // Color of the rect that's drawn if the picture has no texture.
     SDL_Color color = {0, 0, 0, 0};
+
+protected:
+    SDL_BlendMode blendMode;
 };
diff --git a/source/gui/GuiImage.cpp b/source/gui/GuiImage.cpp
index fad3231..c0dc662 100644
--- a/source/gui/GuiImage.cpp
+++ b/source/gui/GuiImage.cpp
@@ -64,3 +64,8 @@ void GuiImage::setTexture(GuiTextureData *tex) {
         this->setSize(tex->getWidth(), tex->getHeight());
     }
 }
+
+int GuiImage::setBlendMode(SDL_BlendMode blendMode) {
+    this->blendMode = blendMode;
+    return this->texture ? this->texture->setBlendMode(blendMode) : 0;
+}
