diff --git a/SDL_FontCache.c b/SDL_FontCache.c
index e534622..b99babb 100644
--- a/SDL_FontCache.c
+++ b/SDL_FontCache.c
@@ -170,6 +170,9 @@ static char* ASCII_STRING = NULL;
 static char* LATIN_1_STRING = NULL;
 static char* ASCII_LATIN_1_STRING = NULL;
 
+// Old color - initially fully transparent
+static SDL_Color oldColor = { 0x00, 0x00, 0x00, 0x00 };
+
 char* FC_GetStringASCII(void)
 {
     if(ASCII_STRING == NULL)
@@ -1751,6 +1754,11 @@ static FC_Rect FC_RenderLeft(FC_Font* font, FC_Target* dest, float x, float y, F
 
 static void set_color_for_all_caches(FC_Font* font, SDL_Color color)
 {
+    if(color.r == oldColor.r && color.g == oldColor.g && color.b == oldColor.b && color.a == oldColor.a)
+        return;
+
+    oldColor = color;
+
     // TODO: How can I predict which glyph caches are to be used?
     FC_Image* img;
     int i;
