diff --git a/SDL_FontCache.h b/SDL_FontCache.h
index 77f5fb4..1b1d612 100644
--- a/SDL_FontCache.h
+++ b/SDL_FontCache.h
@@ -113,13 +113,13 @@ typedef struct FC_GlyphData
 
 // Object creation
 
-FC_Rect FC_MakeRect(float x, float y, float w, float h);
+#define FC_MakeRect(a, b, c, d) (FC_Rect){.x = a, .y = b, .w = c, .h = d}
 
-FC_Scale FC_MakeScale(float x, float y);
+#define FC_MakeScale(a, b) (FC_Scale){.x = a, .y = b}
 
-SDL_Color FC_MakeColor(Uint8 r, Uint8 g, Uint8 b, Uint8 a);
+#define FC_MakeColor(a, b, c, d) (FC_Color){.r = a, .g = b, .b = c, .a = d}
 
-FC_Effect FC_MakeEffect(FC_AlignEnum alignment, FC_Scale scale, SDL_Color color);
+#define FC_MakeEffect(a, b, c) (FC_Effect) { .alignment = a, .scale = s, .color = c }
 
 FC_GlyphData FC_MakeGlyphData(int cache_level, Sint16 x, Sint16 y, Uint16 w, Uint16 h);
 
@@ -186,7 +186,7 @@ void FC_GetUTF8FromCodepoint(char* result, Uint32 codepoint);
 char* U8_alloc(unsigned int size);
 
 /*! Deallocates the given string. */
-void U8_free(char* string);
+#define U8_free(string) free(string)
 
 /*! Allocates a copy of the given string. */
 char* U8_strdup(const char* string);
@@ -201,7 +201,7 @@ int U8_charsize(const char* character);
 int U8_charcpy(char* buffer, const char* source, int buffer_size);
 
 /*! Returns a pointer to the next UTF-8 character. */
-const char* U8_next(const char* string);
+#define U8_next(string) (string + U8_charsize(string))
 
 /*! Inserts a UTF-8 string into 'string' at the given position.  Use a position of -1 to append.  Returns 0 when unable to insert the string. */
 int U8_strinsert(char* string, int position, const char* source, int max_bytes);
