diff --git a/SDL_FontCache.c b/SDL_FontCache.c
index 72483ac..fb156f8 100644
--- a/SDL_FontCache.c
+++ b/SDL_FontCache.c
@@ -154,11 +154,11 @@ static char* replace_concat(char** a, const char* b)
 
 
 // Width of a tab in units of the space width (sorry, no tab alignment!)
-static unsigned int fc_tab_width = 4;
+static const unsigned int fc_tab_width = 4;
 
 // Shared buffer for variadic text
-static char* fc_buffer = NULL;
-static unsigned int fc_buffer_size = 1024;
+static const unsigned int fc_buffer_size = 1024;
+static char fc_buffer[1024];
 
 static Uint8 fc_has_render_target_support = 0;
 
@@ -821,29 +821,12 @@ unsigned int FC_GetBufferSize(void)
     return fc_buffer_size;
 }
 
-void FC_SetBufferSize(unsigned int size)
-{
-    free(fc_buffer);
-    if(size > 0)
-    {
-        fc_buffer_size = size;
-        fc_buffer = (char*)malloc(fc_buffer_size);
-    }
-    else
-        fc_buffer = (char*)malloc(fc_buffer_size);
-}
-
 
 unsigned int FC_GetTabWidth(void)
 {
     return fc_tab_width;
 }
 
-void FC_SetTabWidth(unsigned int width_in_spaces)
-{
-    fc_tab_width = width_in_spaces;
-}
-
 
 
 
@@ -899,9 +882,6 @@ static void FC_Init(FC_Font* font)
 
 	if (font->loading_string == NULL)
 		font->loading_string = FC_GetStringASCII();
-
-    if(fc_buffer == NULL)
-        fc_buffer = (char*)malloc(fc_buffer_size);
 }
 
 static Uint8 FC_GrowGlyphCache(FC_Font* font)
@@ -1477,9 +1457,6 @@ void FC_FreeFont(FC_Font* font)
 
         free(ASCII_LATIN_1_STRING);
         ASCII_LATIN_1_STRING = NULL;
-
-        free(fc_buffer);
-        fc_buffer = NULL;
     }
 }
 
