diff --git a/SDL_FontCache.c b/SDL_FontCache.c
index e534622..1a3d8ce 100644
--- a/SDL_FontCache.c
+++ b/SDL_FontCache.c
@@ -1712,6 +1712,20 @@ static FC_Rect FC_RenderLeft(FC_Font* font, FC_Target* dest, float x, float y, F
         }
 
         codepoint = FC_GetCodepointFromUTF8(&c, 1);  // Increments 'c' to skip the extra UTF-8 bytes
+
+        // Replace characters not displayable by Nintendos font
+        switch(codepoint)
+        {
+            case 0xC58D:
+                codepoint = 'o';
+                break;
+            case 0xC58C:
+                codepoint = 'O';
+                break;
+            case 0xE28091:
+                codepoint = '-';
+        }
+
         if(!FC_GetGlyphData(font, &glyph, codepoint))
         {
             codepoint = ' ';
