diff --git a/SDL_FontCache.c b/SDL_FontCache.c
index 72483ac..7aa9ddd 100644
--- a/SDL_FontCache.c
+++ b/SDL_FontCache.c
@@ -1768,9 +1768,6 @@ FC_Rect FC_Draw(FC_Font* font, FC_Target* dest, float x, float y, const char* fo
         return FC_MakeRect(x, y, 0, 0);
 
     FC_EXTRACT_VARARGS(fc_buffer, formatted_text);
-
-    set_color_for_all_caches(font, font->default_color);
-
     return FC_RenderLeft(font, dest, x, y, FC_MakeScale(1,1), fc_buffer);
 }
 
@@ -2071,8 +2068,6 @@ FC_Rect FC_DrawBox(FC_Font* font, FC_Target* dest, FC_Rect box, const char* form
 
     set_clip(dest, &newclip);
 
-    set_color_for_all_caches(font, font->default_color);
-
     FC_DrawColumnFromBuffer(font, dest, box, NULL, FC_MakeScale(1,1), FC_ALIGN_LEFT);
 
     if(useClip)
@@ -2102,8 +2097,6 @@ FC_Rect FC_DrawBoxAlign(FC_Font* font, FC_Target* dest, FC_Rect box, FC_AlignEnu
         newclip = box;
     set_clip(dest, &newclip);
 
-    set_color_for_all_caches(font, font->default_color);
-
     FC_DrawColumnFromBuffer(font, dest, box, NULL, FC_MakeScale(1,1), align);
 
     if(useClip)
@@ -2133,8 +2126,6 @@ FC_Rect FC_DrawBoxScale(FC_Font* font, FC_Target* dest, FC_Rect box, FC_Scale sc
         newclip = box;
     set_clip(dest, &newclip);
 
-    set_color_for_all_caches(font, font->default_color);
-
     FC_DrawColumnFromBuffer(font, dest, box, NULL, scale, FC_ALIGN_LEFT);
 
     if(useClip)
@@ -2173,6 +2164,7 @@ FC_Rect FC_DrawBoxColor(FC_Font* font, FC_Target* dest, FC_Rect box, SDL_Color c
     else
         set_clip(dest, NULL);
 
+    set_color_for_all_caches(font, font->default_color);
     return box;
 }
 
@@ -2204,6 +2196,7 @@ FC_Rect FC_DrawBoxEffect(FC_Font* font, FC_Target* dest, FC_Rect box, FC_Effect
     else
         set_clip(dest, NULL);
 
+    set_color_for_all_caches(font, font->default_color);
     return box;
 }
 
@@ -2217,8 +2210,6 @@ FC_Rect FC_DrawColumn(FC_Font* font, FC_Target* dest, float x, float y, Uint16 w
 
     FC_EXTRACT_VARARGS(fc_buffer, formatted_text);
 
-    set_color_for_all_caches(font, font->default_color);
-
     FC_DrawColumnFromBuffer(font, dest, box, &total_height, FC_MakeScale(1,1), FC_ALIGN_LEFT);
 
     return FC_MakeRect(box.x, box.y, width, total_height);
@@ -2234,8 +2225,6 @@ FC_Rect FC_DrawColumnAlign(FC_Font* font, FC_Target* dest, float x, float y, Uin
 
     FC_EXTRACT_VARARGS(fc_buffer, formatted_text);
 
-    set_color_for_all_caches(font, font->default_color);
-
     switch(align)
     {
     case FC_ALIGN_CENTER:
@@ -2263,8 +2252,6 @@ FC_Rect FC_DrawColumnScale(FC_Font* font, FC_Target* dest, float x, float y, Uin
 
     FC_EXTRACT_VARARGS(fc_buffer, formatted_text);
 
-    set_color_for_all_caches(font, font->default_color);
-
     FC_DrawColumnFromBuffer(font, dest, box, &total_height, scale, FC_ALIGN_LEFT);
 
     return FC_MakeRect(box.x, box.y, width, total_height);
@@ -2284,7 +2271,9 @@ FC_Rect FC_DrawColumnColor(FC_Font* font, FC_Target* dest, float x, float y, Uin
 
     FC_DrawColumnFromBuffer(font, dest, box, &total_height, FC_MakeScale(1,1), FC_ALIGN_LEFT);
 
-    return FC_MakeRect(box.x, box.y, width, total_height);
+    FC_Rect ret = FC_MakeRect(box.x, box.y, width, total_height);
+    set_color_for_all_caches(font, font->default_color);
+    return ret;
 }
 
 FC_Rect FC_DrawColumnEffect(FC_Font* font, FC_Target* dest, float x, float y, Uint16 width, FC_Effect effect, const char* formatted_text, ...)
@@ -2313,7 +2302,9 @@ FC_Rect FC_DrawColumnEffect(FC_Font* font, FC_Target* dest, float x, float y, Ui
 
     FC_DrawColumnFromBuffer(font, dest, box, &total_height, effect.scale, effect.alignment);
 
-    return FC_MakeRect(box.x, box.y, width, total_height);
+    FC_Rect ret = FC_MakeRect(box.x, box.y, width, total_height);
+    set_color_for_all_caches(font, font->default_color);
+    return ret;
 }
 
 static FC_Rect FC_RenderCenter(FC_Font* font, FC_Target* dest, float x, float y, FC_Scale scale, const char* text)
@@ -2389,8 +2380,6 @@ FC_Rect FC_DrawScale(FC_Font* font, FC_Target* dest, float x, float y, FC_Scale
 
     FC_EXTRACT_VARARGS(fc_buffer, formatted_text);
 
-    set_color_for_all_caches(font, font->default_color);
-
     return FC_RenderLeft(font, dest, x, y, scale, fc_buffer);
 }
 
@@ -2401,8 +2390,6 @@ FC_Rect FC_DrawAlign(FC_Font* font, FC_Target* dest, float x, float y, FC_AlignE
 
     FC_EXTRACT_VARARGS(fc_buffer, formatted_text);
 
-    set_color_for_all_caches(font, font->default_color);
-
     FC_Rect result;
     switch(align)
     {
@@ -2432,7 +2419,9 @@ FC_Rect FC_DrawColor(FC_Font* font, FC_Target* dest, float x, float y, SDL_Color
 
     set_color_for_all_caches(font, color);
 
-    return FC_RenderLeft(font, dest, x, y, FC_MakeScale(1,1), fc_buffer);
+    FC_Rect ret = FC_RenderLeft(font, dest, x, y, FC_MakeScale(1,1), fc_buffer);
+    set_color_for_all_caches(font, font->default_color);
+    return ret;
 }
 
 
@@ -2462,6 +2451,7 @@ FC_Rect FC_DrawEffect(FC_Font* font, FC_Target* dest, float x, float y, FC_Effec
             break;
     }
 
+    set_color_for_all_caches(font, font->default_color);
     return result;
 }
 
