diff --git a/SDL_FontCache.c b/SDL_FontCache.c
index e534622..a7da7f3 100644
--- a/SDL_FontCache.c
+++ b/SDL_FontCache.c
@@ -496,8 +496,9 @@ char* U8_strdup(const char* string)
     if(string == NULL)
         return NULL;
 
-    result = (char*)malloc(strlen(string)+1);
-    strcpy(result, string);
+    size_t len = strlen(string) + 1;
+    result = (char*)malloc(len);
+    memcpy(result, string, len);
 
     return result;
 }
